### PR TITLE
atlas console without dropwizard dependency

### PR DESCRIPTION
**Pull request by rjullman**
Have console read config files without the dropwizard dependency.

### CODE REVIEW

File: atlasdb-console/src/main/groovy/com/palantir/atlasdb/console/module/AtlasCoreModule.groovy [hunk 1]
```diff
@@ -135,10 +136,12 @@ class AtlasCoreModule implements AtlasConsoleModule {
     }
 
     public connect(String yamlFilePath = null) {
```
Comment (jboreiko) [inline atlasdb-console/src/main/groovy/com/palantir/atlasdb/console/module/AtlasCoreModule.groovy]: Looks like we should be able to consolidate the logic between these two 'connect / config load' methods.  A simple solution might be to have the config object have static method for loading from yaml.

File: atlasdb-cli/build.gradle [hunk 2]
```diff
@@ -11,6 +11,11 @@ dependencies {
   compile project(':atlasdb-impl-shared')
   compile 'io.airlift:airline:0.7'
   compile 'com.google.dagger:dagger:2.0.2'
+  compile 'com.fasterxml.jackson.core:jackson-databind:' + libVersions.jackson
+  compile 'com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:' + libVersions.jackson
+  compile 'io.dropwizard:dropwizard-jackson:' + libVersions.dropwizard
```
Comment (joelea) [inline atlasdb-cli/build.gradle]: I assume this is to pull in the service discovery stuff?

File: atlasdb-console/src/main/groovy/com/palantir/atlasdb/console/module/AtlasCoreModule.groovy (hunk 1 above)
Comment (joelea) [inline atlasdb-console/src/main/groovy/com/palantir/atlasdb/console/module/AtlasCoreModule.groovy]: That seems reasonable.

File: atlasdb-cli/build.gradle (hunk 2 above)
Comment (rjullman) [inline atlasdb-cli/build.gradle]: yes:
(1) databind for handling JsonNodes (we can potentially avoid this by making a jackson object that looks like the Dropwizard configuration object that doesn't extend from dropwizard)
(2) yaml for reading the config
(3) dropwizard-jackson for having the service discovery object mapper plugin

File: atlasdb-console/src/main/groovy/com/palantir/atlasdb/console/module/AtlasCoreModule.groovy (hunk 1 above)
Comment (rjullman) [inline atlasdb-console/src/main/groovy/com/palantir/atlasdb/console/module/AtlasCoreModule.groovy]: sounds good (y) -- new pr coming

**Comment by joelea** [comment]
This looks sensible with John's change added.

File: atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbConfigs.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+public final class AtlasDbConfigs {
+
+    public static final String ATLASDB_CONFIG_ROOT = "atlasdb";
+
+    private AtlasDbConfigs() {
+        // uninstantiable
+    }
+
+    public static AtlasDbConfig load(File configFile) throws IOException {
```
Comment (jboreiko) [inline atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbConfigs.java]: dope, this is secretly what I wanted all along :)

**Pull request merged**